src/ert/gui/model/real_list.py

```diff
@@ -107,91 +90,71 @@ def parent(self, child: Optional[QModelIndex] = None) -> Optional[QObject]:
+                return self.index(src_index.row(), src_index.column(), QModelIndex())
+        return QModelIndex()
 
     def _source_data_changed(
         self, top_left: QModelIndex, bottom_right: QModelIndex, roles: List[int]
     ) -> None:
-        if top_left.internalPointer() is None:
-            return
-        if not top_left.data(IsRealizationRole):
-            return
-        proxy_top_left = self.mapFromSource(top_left)
-        proxy_bottom_right = self.mapFromSource(bottom_right)
-        if not proxy_top_left.isValid() or not proxy_bottom_right.isValid():
-            return
-        self.dataChanged.emit(proxy_top_left, proxy_bottom_right, roles)
+        if top_left.internalPointer() and top_left.data(IsRealizationRole):
+            proxy_top_left = self.mapFromSource(top_left)
+            proxy_bottom_right = self.mapFromSource(bottom_right)
+
+            if all([proxy_top_left.isValid(), proxy_bottom_right.isValid()]):
+                self.dataChanged.emit(proxy_top_left, proxy_bottom_right, roles)
 
     def _source_rows_about_to_be_inserted(
         self, parent: QModelIndex, start: int, end: int
     ) -> None:
-        if not parent.isValid():
-            return
-        if not self._accept_index(parent):
-            return
-        self.beginInsertRows(self.mapFromSource(parent), start, end)
+        if parent.isValid() and self._accept_index(parent):
+            self.beginInsertRows(self.mapFromSource(parent), start, end)
 
     def _source_rows_inserted(
         self, parent: QModelIndex, _start: int, _end: int
     ) -> None:
-        if not parent.isValid():
-            return
-        if not self._accept_index(parent):
-            return
-        self.endInsertRows()
+        if parent.isValid() and self._accept_index(parent):
+            self.endInsertRows()
 
     def _accept_index(self, index: QModelIndex) -> bool:
-        if index.internalPointer() is None:
-            return False
         # If the index under test isn't a realization, it is of no interest as
         # this model should only consist of realization indices.
-        if not index.data(IsRealizationRole):
-            return False
-
-        # traverse upwards the tree, checking whether this index is accepted or
-        # not.
-        while index.isValid() and index.internalPointer() is not None:
-            if index.data(IsEnsembleRole) and index.row() != self._iter:
-                return False
-            index = index.parent()
-        return True
+        if index.internalPointer() and index.data(IsRealizationRole):
+            # traverse upwards the tree, checking whether this index is accepted or not
+            while index.isValid() and index.internalPointer():
+                if index.data(IsEnsembleRole) and index.row() != self._iter:
+                    return False
+                index = index.parent()
+            return True
+        return False
```

Comment:
same as with _accept_index in job_list.py. would prefer less nesting.

Reply:
Same as the other one, let's go with the less nesting. 👍